helix-core/src/test/java/org/apache/helix/integration/TestZkCallbackHandlerLeak.java

```diff
@@ -162,33 +154,25 @@ public boolean verify() throws Exception {
     Assert.assertTrue(result);
 
     // check controller zk-watchers
-    result = TestHelper.verify(new TestHelper.Verifier() {
-
-      @Override
-      public boolean verify() throws Exception {
-        Map<String, Set<String>> watchers = ZkTestHelper.getListenersBySession(ZK_ADDR);
-        Set<String> watchPaths = watchers.get("0x" + controller.getSessionId());
-        // System.out.println("controller watch paths after session expiry: " + watchPaths);
-
-        // where n is number of nodes and r is number of resources
-        // one participant is disconnected, and its task current states are removed
-        return watchPaths.size() == (8 + r + (6 + r + taskResourceCount) * (n - 1) + 6 + r);
-      }
+    result = TestHelper.verify(() -> {
+      Map<String, Set<String>> watchers = ZkTestHelper.getListenersBySession(ZK_ADDR);
+      Set<String> watchPaths = watchers.get("0x" + controller.getSessionId());
+      // System.out.println("controller watch paths after session expiry: " + watchPaths);
```

Comment:
Do we want to leave this print statement?

Reply:
Removed